en/identity-server/next/docs/references/extend/federation/write-a-custom-federated-authenticator.md

```diff
@@ -0,0 +1,304 @@
+          /**
+           * Check whether the authentication or logout request can be handled by the
+           * authenticator
+           * 
+           * @param request
+           * @return boolean
+           */
+          public boolean canHandle(HttpServletRequest request);
+          
+          /**
+           * Process the authentication or logout request.
+           * 
+           * @param request
+           * @param response
+           * @param context
+           * @return the status of the flow
+           * @throws AuthenticationFailedException
+           * @throws LogoutFailedException
+           */
+          public AuthenticatorFlowStatus process(HttpServletRequest request,
+                  HttpServletResponse response, AuthenticationContext context)
+                  throws AuthenticationFailedException, LogoutFailedException;
+          
+          /**
+           * Get the Context identifier sent with the request. This identifier is used
+           * to retrieve the state of the authentication/logout flow
+           * 
+           * @param request
+           * @return
+           */
+          public String getContextIdentifier(HttpServletRequest request);
+          
+          /**
+           * Get the name of the Authenticator
+           * @return name
+           */
+          public String getName();
+          
+          /**
+           * @return
+           */
+          public String getFriendlyName();
+          
+          /**
+           * Get the claim dialect URI if this authenticator receives claims in a standard dialect
+           * and needs to be mapped to the Carbon dialect http://wso2.org/claims
+           * @return boolean
+           */
+          public String getClaimDialectURI();
+          
+          /**
+           * @return
+           */
+          public List<Property> getConfigurationProperties();
+      }
+    ```
+
+---
+
+## Write the federated authenticator
+
+To write the federated authenticator:
+
+1. Create a `maven` project for the custom federated authenticator. Refer the following files:
+    - The [pom.xml](https://github.com/wso2/samples-is/blob/master/authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/pom.xml) file used for the sample custom federated authenticator.
+    - The [service component class](https://github.com/wso2/samples-is/blob/master/authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/src/main/java/org/wso2/carbon/identity/custom/federated/authenticator/internal/CustomFederatedAuthenticatorServiceComponent.java) file to deploy in WSO2 Identity Server and register it as a federated authenticator.
+
+2. Write the [custom federated authenticator](https://github.com/wso2/samples-is/blob/master/authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/src/main/java/org/wso2/carbon/identity/custom/federated/authenticator/CustomFederatedAuthenticator.java) by extending the
```

Comment:
Better if we could point to a compatible released tag's source code instead of linking to main branch

Reply:
Check other places as well, where we have linked to master/main branch

Reply:
Fixed with ffc1d9c4db8dd9d02f4564c83965cfbdd6f16529